src/NifFile.cpp

```diff
@@ -949,455 +949,451 @@ void NifFile::Optimize() {
 OptResult NifFile::OptimizeFor(OptOptions& options) {
 	OptResult result;
 
-	if (options.targetVersion.IsSSE() && hdr.GetVersion().IsSK()) {
-		// SK -> SSE
-		if (!isTerrain)
-			result.dupesRenamed = RenameDuplicateShapes();
+	const bool toSSE = options.targetVersion.IsSSE() && hdr.GetVersion().IsSK();
+	const bool toLE = options.targetVersion.IsSK() && hdr.GetVersion().IsSSE();
 
-		hdr.SetVersion(options.targetVersion);
+	if (!toSSE && !toLE) {
+		result.versionMismatch = true;
+		return result;
+	}
+
+	if (!isTerrain)
+		result.dupesRenamed = RenameDuplicateShapes();
 
-		for (auto& shape : GetShapes()) {
+	hdr.SetVersion(options.targetVersion);
+
+	auto shapes = GetShapes();
+	if (toSSE) {
+		std::for_each(shapes.begin(), shapes.end(), [this, &options, &result](auto* shape) {
```

Comment:
What's the reason for the std::for_each loop?

Reply:
For now, none. I added it while trying to add multithreading, it's a leftover. I can go back to a normal for loop if you prefer, it's almost the same

Reply:
I'd prefer that, since right now it looks out of place compared to all the other auto-for-loops.